modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/configuration/DistributionZonesHighAvailabilityConfiguration.java

```diff
@@ -35,22 +36,28 @@ public class DistributionZonesHighAvailabilityConfiguration {
     static final String PARTITION_DISTRIBUTION_RESET_TIMEOUT = "partitionDistributionResetTimeout";
 
     /** Default value for the {@link #PARTITION_DISTRIBUTION_RESET_TIMEOUT}. */
-    private static final long PARTITION_DISTRIBUTION_RESET_TIMEOUT_DEFAULT_VALUE = 0;
+    private static final int PARTITION_DISTRIBUTION_RESET_TIMEOUT_DEFAULT_VALUE = 0;
 
     private final SystemDistributedConfiguration systemDistributedConfig;
 
     /** Determines partition group reset timeout after a partition group majority loss. */
-    private volatile long partitionDistributionResetTimeout;
+    private volatile int partitionDistributionResetTimeout;
+
+    /** Listener, which receives (timeout, revision) on every configuration update. */
+    private final BiConsumer<Integer, Long> partitionDistributionResetListener;
 
     /** Constructor. */
-    public DistributionZonesHighAvailabilityConfiguration(SystemDistributedConfiguration systemDistributedConfig) {
+    public DistributionZonesHighAvailabilityConfiguration(
+            SystemDistributedConfiguration systemDistributedConfig,
+            BiConsumer<Integer, Long> partitionDistributionResetListener) {
```

Comment:
I'd rather add listener as a separate method instead of propagating it into constructor + start(). Precisely I mean adding `onPartitionDistributionResetTimeoutUpdate()`. In that case DistributionZonesHighAvailabilityConfiguration will have clear contract

- partitionDistributionResetTimeout()
- onPartitionDistributionResetTimeoutUpdate()

Reply:
Not sure that I understand: onPartitionDistributionResetTimeoutUpdate() must be a getter for the listener? For what purpose?  Moreover we have no use cases for the mutating of the listener, so I wanted to highlight it in the contract - no any setters or getters, just the argument of the constructor

Reply:
> Not sure that I understand: onPartitionDistributionResetTimeoutUpdate() must be a getter for the listener?
No, I mean `onPartitionDistributionResetTimeoutUpdate(listener)`

> Moreover we have no use cases for the mutating of the listener, so I wanted to highlight it in the contract - no any setters or getters, just the argument of the constructor

Well, it's a valid point. I still believe that listener() is cleaner then constructor + start here though. Up to you.

Reply:
I would keep it this way, until the time when we will have (maybe) many listeners for this event and extend API by the listeners mutations.

Reply:
Ok.